Ports/SDL_mixer/patches/0001-libtool-Enable-shared-library-support-for-SerenityOS.patch

```diff
@@ -0,0 +1,76 @@
+From bdb6569ba8bcc1332ee220b3128e94c8c69d47f7 Mon Sep 17 00:00:00 2001
+From: Tim Schumacher <timschumi@gmx.de>
+Date: Sun, 29 May 2022 15:01:28 +0200
+Subject: [PATCH] libtool: Enable shared library support for SerenityOS
+
+For some odd reason, libtool handles the configuration for shared
+libraries entirely statically and in its configure script. If no
+shared library support is "present", building shared libraries is
+disabled entirely.
+
+Fix that by just adding the appropriate configuration options for
+`serenity`. This allows us to finally create dynamic libraries
+automatically using libtool, without having to manually link the
+static library into a shared library.
+---
+ configure | 23 +++++++++++++++++++++++
+ 1 file changed, 23 insertions(+)
+
+diff --git a/configure b/configure
+index e7c8c97..37ef6b5 100755
+--- a/configure
++++ b/configure
+@@ -4244,6 +4244,10 @@ sysv4 | sysv4.3*)
+ tpf*)
+   lt_cv_deplibs_check_method=pass_all
+   ;;
++
++serenity*)
++  lt_cv_deplibs_check_method=pass_all
++  ;;
+ esac
+ 
+ fi
+@@ -7375,6 +7379,10 @@ echo $ECHO_N "checking for $compiler option to produce PIC... $ECHO_C" >&6; }
+       lt_prog_compiler_static='-Bstatic'
+       ;;
+ 
++    serenity*)
++      lt_prog_compiler_can_build_shared=yes
++      ;;
++
+     *)
+       lt_prog_compiler_can_build_shared=no
+       ;;
+@@ -8687,6 +8695,10 @@ rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
+       hardcode_shlibpath_var=no
+       ;;
+ 
++    serenity*)
++      ld_shlibs=yes
++      ;;
++
+     *)
+       ld_shlibs=no
+       ;;
+@@ -9617,6 +9629,17 @@ uts4*)
+   shlibpath_var=LD_LIBRARY_PATH
+   ;;
+ 
++serenity*)
++  version_type=linux
++  need_lib_prefix=no
++  need_version=no
++  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}${major} ${libname}${shared_ext}'
```

Comment:
This is something I just noticed now but it would annoy me to no end if it actually ends up being merged. Would you mind quickly editing the patch to add curly braces around `versuffix` as well? :^)

Reply:
As an aside, we should really just add ourselves to libtool.

Would fix #6694.

Reply:
For newer versions sure, I guess we are stable enough now to allow for that. But that sadly doesn't fix old versions of `libtool` (`use_fresh_libtool` wen?).

Reply:
> `use_fresh_libtool` wen

That just sounds like `autoconf` hell. Trying to debug why libmpg123 fails to reconfigure itself on some systems was painful enough...